src/lib/Siteaccess/SiteaccessResolver.php

```diff
@@ -42,33 +60,13 @@ public function getSiteaccessesForLocation(
 
         $eligibleSiteaccesses = [];
         foreach ($this->getSiteaccesses() as $siteaccess) {
-            $rootLocationId = $this->configResolver->getParameter(
-                'content.tree_root.location_id',
-                null,
-                $siteaccess
-            );
-            if (!in_array($rootLocationId, $location->path)) {
-                continue;
+            $context = new SiteaccessPreviewVoterContext($location, $siteaccess, $languageCode, $contentLanguages);
```

Comment:
I think it would make more sense to pass `$contentInfo` and `$versionInfo` instead of `$contentLanguages` as it gives more flexibility.

Reply:
I will pass versionInfo because contentInfo is in location